bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/parallacticangle/ParallacticAngleControls.scala

```diff
@@ -161,13 +165,34 @@ class ParallacticAngleControls(isPaUi: Boolean) extends GridBagPanel with Publis
     } {
       val spObs = ispObs.getDataObject.asInstanceOf[SPObservation]
       val sameNight = spObs.getSchedulingBlock.asScalaOpt.exists(_.sameObservingNightAs(sb))
-      spObs.setSchedulingBlock(ImOption.apply(sb))
-      ispObs.setDataObject(spObs)
-      if (!sameNight) {
-        EphemerisUpdater.unsafeRefreshEphemerides(ispObs, e.getWindow)
-      }
-      callback.run()
-      resetComponents()
+
+      // IO action on EDT
+      def edt[A](a: => Unit): IO[Unit] =
+        IO(Swing.onEDT(a))
+
+      // The update we want to do
+      val action: IO[Unit] =
+        for {
+          _ <- IO(spObs.setSchedulingBlock(ImOption.apply(sb)))
+          _ <- IO(ispObs.setDataObject(spObs))
+          _ <- sameNight.unlessM(EphemerisUpdater.refreshEphemerides(ispObs, e.getWindow))
+          _ <- edt(callback.run())
+          _ <- edt(resetComponents())
+        } yield ()
+
+      // Without sending events
+      val quiet: IO[Unit] =
+        IO(ispObs.setSendingEvents(false)) *> action ensuring IO(ispObs.setSendingEvents(true))
```

Comment:
Yes, BAGS and the OT generally are watching for change events on the observation and everything gets interleaved when that happens because I'm making potentially a bunch of updates. This essentially makes the update atomic.